### PR TITLE
Add Poetry Build

### PR DESCRIPTION
**Pull request by talarian1**
- [x] All [tests](https://github.com/jfrog/build-info-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/build-info-go/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

File: utils/pythonutils/poetryutils.go [hunk 1]
```diff
@@ -116,9 +125,105 @@ func extractPackagesFromPoetryLock(lockFilePath string) (dependencies map[string
...
+	_, dependenciesVersions, err := extractPackagesFromPoetryLock(filePath)
+	if err != nil {
+		return nil, err
+	}
+	dependenciesFiles = map[string]entities.Dependency{}
+	for dependency, version := range dependenciesVersions {
+		directUrlPath := fmt.Sprintf("%s%s-%s.dist-info%sdirect_url.json", sitePackagesPath, dependency, version, string(os.PathSeparator))
+		directUrlFile, err := ioutil.ReadFile(directUrlPath)
+		if err != nil {
+			log.Debug(fmt.Sprintf("Could not resolve download path for package: %s, continuing...", dependency))
```
Comment (eyalbe4) [inline utils/pythonutils/poetryutils.go:156]: Let's also add the error string into this debug log message.

File: utils/pythonutils/poetryutils.go [hunk 2]
```diff
@@ -116,9 +125,105 @@ func extractPackagesFromPoetryLock(lockFilePath string) (dependencies map[string
...
+			continue
+		}
+		directUrl := packagedDirectUrl{}
+		err = json.Unmarshal(directUrlFile, &directUrl)
+		if err != nil {
+			log.Debug(fmt.Sprintf("Could not resolve download path for package: %s, continuing...", dependency))
```
Comment (eyalbe4) [inline utils/pythonutils/poetryutils.go:162]: Let's also add the error string into this debug log message.

File: utils/pythonutils/poetryutils.go [hunk 3]
```diff
@@ -116,9 +125,105 @@ func extractPackagesFromPoetryLock(lockFilePath string) (dependencies map[string
...
 	for _, dependency := range poetryLockFile.Package {
-		dependenciesVersions[dependency.Name] = dependency.Version
+		dependenciesVersions[strings.ToLower(dependency.Name)] = dependency.Version
 		dependencyName := dependency.Name + ":" + dependency.Version
 		dependencies[dependencyName] = maps.Keys(dependency.Dependencies)
 	}
 	return
 }
+
+func extractPoetryDependenciesFiles(srcPath string, cmdArgs []string, log utils.Log) (dependenciesFiles map[string]entities.Dependency, err error) {
```
Comment (eyalbe4) [inline utils/pythonutils/poetryutils.go:135]: * Let's add a description for this function, specifically describing the returned values and the structure of `dependenciesFiles`.
* Let's add a unit test for this function.
* I also suggest adding some more focused tests for the `getSitePackagesPath` function.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: utils/pythonutils/poetryutils.go (hunk 3 above)
Comment (talarian1) [reply utils/pythonutils/poetryutils.go:135]: Tests will be added in following PR.

**Pull request merged**